src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as tree from './tree_methods'
+import * as _ from 'lodash'
+
+let root;
+
+let treeRoot = new tree.TreeNode("korzen", false);
+let note1 = new tree.TreeNode("notka1", false);
+let note2 = new tree.TreeNode("notka2", false);
+let note3 = new tree.TreeNode("notka3", false);
+let note21 = new tree.TreeNode("notka1od2", false);
+let note22 = new tree.TreeNode("notka2od2", false);
+let note221 = new tree.TreeNode("notka1od22", false);
+let note222 = new tree.TreeNode("notka2od22", false);
+let note31 = new tree.TreeNode("notka1od3", false);
+
+treeRoot.add(note1);
```

Comment:
Can we do fluent builder here or is there a reason for not returning self on add?

Reply:
no idea what it means